integration-tests/src/test/java/oracle/kubernetes/operator/ItInitContainers.java

```diff
@@ -0,0 +1,369 @@
+    logger.info(modYaml);
+    testInitContainer(modYaml);
+    String cmd = "kubectl get pod " + adminPodName + " -n " + domain.getDomainNs();
+    TestUtils.checkCmdInLoop(cmd, "Init:CrashLoopBackOff", adminPodName);
+    logger.log(Level.INFO, "SUCCESS - {0}", testMethodName);
+  }
+
+  /**
+   * Add initContainers at domain and admin server level and verify init container runs at both
+   * level when the names are different
+   *
+   * @throws Exception when domain.yaml cannot be read or modified to include the initContainers or
+   *     weblogic server pod doesn't go through initialization and ready state
+   */
+  @Test
+  public void testInitContainerDiffLevelDiffName() throws Exception {
+    Assume.assumeFalse(QUICKTEST);
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+    String pods[] = {domainUid + "-" + domain.getAdminServerName(), domainUid + "-managed-server2"};
+
+    // Modify the original domain yaml to include restartVersion in admin server node
+    DomainCrd crd = new DomainCrd(originalYaml);
+    crd.addInitContNode("spec", null, null, "busybox1", "sleep");
+    crd.addInitContNode("adminServer", null, null, "busybox2", "sleep");
+    String modYaml = crd.getYamlTree();
+    logger.info(modYaml);
+    testInitContainer(modYaml);
+    String cmd = "kubectl get pod " + pods[0] + " -n " + domain.getDomainNs();
+
+    TestUtils.checkCmdInLoop(cmd, "Init:0/2", pods[0]);
+    TestUtils.checkCmdInLoop(cmd, "Init:1/2", pods[0]);
+    TestUtils.checkPodReady(pods[0], domain.getDomainNs());
+
+    logger.info("Verifying if the pods are recreated with initialization");
+    verifyPodInitialized(pods[1]);
+    logger.log(Level.INFO, "SUCCESS - {0}", testMethodName);
+  }
+
+  /**
+   * Add initContainers at domain and admin server level and verify init container is not run at
+   * both level when the names are same
+   *
+   * @throws Exception when domain.yaml cannot be read or modified to include the initContainers or
+   *     weblogic server pod doesn't go through initialization and ready state
+   */
+  @Test
+  public void testInitContainerDiffLevelSameName() throws Exception {
+    Assume.assumeFalse(QUICKTEST);
+    String testMethodName = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+    String pods[] = {domainUid + "-" + domain.getAdminServerName(), domainUid + "-managed-server2"};
+
+    // Modify the original domain yaml to include restartVersion in admin server node
+    DomainCrd crd = new DomainCrd(originalYaml);
+    crd.addInitContNode("spec", null, null, "busybox", "foo");
+    crd.addInitContNode("adminServer", null, null, "busybox", "sleep");
+    crd.addInitContNode("clusters", "cluster-1", null, "busybox", "sleep");
+    String modYaml = crd.getYamlTree();
+    logger.info(modYaml);
+    testInitContainer(modYaml);
+    for (String pod : pods) {
```

Comment:
you may need to check  TestUtils.checkCmdInLoop(cmd, "Init:0/1", pod[0]); to make sure only one initcontainer is started